File: results-05-12/typescript/typescript-00016-benchmark-file-store/qwen2-5-coder-32b/extracted-src-code.ts
import { getHomePath } from './shinkai-local-support.ts';

type CONFIG = {};
type INPUTS = { url: string };
type OUTPUT = { filePath: string };

export async function run(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
  const response = await fetch(inputs.url);
  if (!response.ok) {
    throw new Error(`Failed to download image from ${inputs.url}`);
  }
  const data = await response.arrayBuffer();
  const filePath = `${getHomePath()}/downloaded_image.png`;
  await Deno.writeFile(filePath, new Uint8Array(data));
  return { filePath };
}

File: results-05-12/python/python-00008-benchmark-download-website/qwen2-5-coder-32b/extracted-src-code.py
from typing import Dict, Any, Optional
import requests

class CONFIG:
    pass

class INPUTS:
    url: str

class OUTPUT:
    content: str

async def run(config: CONFIG, inputs: INPUTS) -> OUTPUT:
    response = requests.get(inputs.url)
    output = OUTPUT()
    output.content = response.text
    return output